examples/howto/server_embed/django_embed/django_embed/views.py

```diff
@@ -0,0 +1,76 @@
+from os.path import join
+from typing import Any
+
+from django.http import HttpRequest, HttpResponse
+from django.conf import settings
+from django.shortcuts import render
+
+from bokeh.plotting import figure
+from bokeh.layouts import column
+from bokeh.models import ColumnDataSource, Slider
+from bokeh.document import Document
+from bokeh.themes import Theme
+from bokeh.embed import server_document
+
+from bokeh.sampledata.sea_surface_temperature import sea_surface_temperature
+
+from .shape_viewer import shape_viewer
+
+theme = Theme(filename=join(settings.THEMES_DIR, "theme.yaml"))
+
+def shape_viewer_handler(doc: Document) -> None:
+    panel = shape_viewer()
+    panel.server_doc(doc)
+
+def sea_surface_handler(doc: Document) -> None:
+    df = sea_surface_temperature.copy()
+    source = ColumnDataSource(data=df)
+
+    plot = figure(x_axis_type="datetime", y_range=(0, 25), y_axis_label="Temperature (Celsius)",
+                  title="Sea Surface Temperature at 43.18, -70.43")
+    plot.line("time", "temperature", source=source)
+
+    def callback(attr: str, old: Any, new: Any) -> None:
+        if new == 0:
+            data = df
+        else:
+            data = df.rolling("{0}D".format(new)).mean()
+        source.data = ColumnDataSource(data=data).data
+
+    slider = Slider(start=0, end=30, value=0, step=1, title="Smoothing by N Days")
+    slider.on_change("value", callback)
+
+    doc.theme = theme
+    doc.add_root(column(slider, plot))
+
+def with_request(f):
+    def wrapper(doc):
+        return f(doc, doc.session_context.request)
+    return wrapper
+
+@with_request
+def sea_surface_handler_with_template(doc: Document, request: Any) -> None:
```

Comment:
@swainn, after all, the only reasonable way of accessing the request (or scope in channels' terms) was to make it available as a part of the document (`doc.session_context.request`), at least without bigger changes to bokeh server. With a simple decorator this can be nicely hidden away, like in this example. I can make this decorator a part of django server API if we settle on this approach.

Reply:
@mattpap this will work for us. The decorator is great for convenience, but I don't mind accessing it on the `doc` object directly either.